### DISCUSSION TITLE
Disable piston

### DISCUSSION THREAD

**Discussion by APonder-Dev** [GriefPrevention/GriefPrevention#2047] [Q&A]
Is there a way we can disable pistons altogether?

**Comment by Jikoo** [comment]
I feel like that's outside of the scope of GP - GP is trying to prevent grief to player structures.

If you're having issues with players building contraptions that are lagging your server, you should probably look into using a more optimized redstone algorithm (if you're using Paper or a fork, [Eigencraft and Alternating Current are available](https://docs.papermc.io/paper/reference/world-configuration#redstone-implementation)).

You could also try to find a redstone-limiting plugin, but be aware that they often fall into the category of ["too good to be true"](https://github.com/YouHaveTrouble/minecraft-optimization#too-good-to-be-true-plugins) and cause more harm (and inconvenience to players) than good.

Alternately, write your own plugin:
```java
@EventHandler
void onPistonExtend(@NotNull BlockPistonExtendEvent event) {
  // Cancel event.
  event.setCancelled(true);
  // Remove piston.
  event.getBlock().setType(Material.AIR);
}
// Note: Not bothering with piston retraction because only existing pistons would be able to retract;
// they will be removed as soon as they attempt to extend again.
```
You could even do things like remove the piston crafting recipes and block piston placement.
